sky/backends/backend_utils.py

```diff
@@ -1831,9 +1831,9 @@ def check_cluster_name_is_valid(cluster_name: str,
         if len(cluster_name) > _MAX_CLUSTER_NAME_LEN_FOR_GCP:
             with ux_utils.print_exception_no_traceback():
                 raise ValueError(
-                    f'Cluster name {cluster_name!r} has {len(cluster_name)}'
-                    f' chars; maximum length is {_MAX_CLUSTER_NAME_LEN_FOR_GCP}'
-                    ' chars.')
+                    f'ClusterNameTooLong: Cluster name {cluster_name!r} has '
+                    f'{len(cluster_name)} chars; maximum length is '
```

Comment:
nit, not necessary to fix: I prefer creating a new exception class (e.g. `ClusterNameError`) when the error will be surfaced to the user and/or is meant to be caught elsewhere in the code. That way we don't have to use hardcoded strings to catch this specific exception and looks nicer to the user too :)

Reply:
Good point! I renamed the exceptions to `InvalideClusterNameError`.

Tested:
- [ ] `sky spot launch -n i-am-a-very-long-name-too-make-it-fail --cloud gcp 'sleep 10'`
- [ ] `sky spot launch -n i-am-a-very-long-name-too-make-it-fail --cloud gcp 'sleep 10' --retry-until-up`